src/node_api.cc

```diff
@@ -935,18 +957,21 @@ class ThreadSafeFunction : public node::AsyncResource {
   // These methods must only be called from the loop thread.
 
   napi_status Init() {
+    uv_loop_t* loop = nullptr;
```

Comment:
> If this is going to crash on any error, why not break the abstraction and directly call `env->node_env()->event_loop();`

It’s not going to crash, which is why there’s a `CHECK` :) But since we have an N-API function for exactly this purpose, I think it makes sense to use it here.

> (I'd go as far as `const uv_loop_t& loop = *(env->node_env()->event_loop());`, but I know I'm pushing it :)

We can’t use a `const uv_loop_t&` :)